runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp

```diff
@@ -114,6 +136,17 @@ initImpl(J9VMThread *currentThread, j9object_t membernameObject, j9object_t refO
 
 		J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(methodID->method);
 
+		if (isPolymorphicMHMethod(currentThread->javaVM, J9_CLASS_FROM_METHOD(methodID->method), J9ROMMETHOD_NAME(romMethod))
+#if JAVA_SPEC_VERSION >= 9
+		|| ((J9_CLASS_FROM_METHOD(methodID->method) == J9VMJAVALANGINVOKEVARHANDLE(currentThread->javaVM))
+			&& VM_VMHelpers::isPolymorphicVarHandleMethod(J9UTF8_DATA(J9ROMMETHOD_NAME(romMethod)),J9UTF8_LENGTH(J9ROMMETHOD_NAME(romMethod)))
+			)
+#endif
+		) {
+			/* do not initialize polymorphic MH/VH methods */
+			return;
```

Comment:
Where do they get initialized now?  Do we not need the `flags` or declaringClass or....?

Reply:
`MethodHandleNatives.init` returns with `MemberName.clazz == null`. Then, `MemberName` gets initialized in Java code: https://github.com/ibmruntimes/openj9-openjdk-jdk16/blob/a6d3f24ce5f8ec72bf3172f24833c2108ebc0dce/src/java.base/share/classes/java/lang/invoke/MemberName.java#L585-L608.

Reply:
uninitialized polymorphic MemberName will be handled in java code at https://github.com/ibmruntimes/openj9-openjdk-jdk16/blob/openj9/src/java.base/share/classes/java/lang/invoke/MemberName.java#L586

Reply:
It would be good to call that out in the comment here - `Do not initialize polymorphic MH/VH methods as the Java code handles the "MemberName.clazz == null" case for them`

Reply:
updated comment